metis-core/metis-core-service/src/main/java/eu/europeana/metis/core/execution/WorkflowPostProcessor.java

```diff
@@ -50,49 +66,91 @@ public WorkflowPostProcessor(DepublishRecordIdDao depublishRecordIdDao,
+      depublishDatasetPostProcess(datasetId);
+    } else {
+      depublishRecordPostProcess(plugin, datasetId);
+    }
+  }
+
+  private void depublishDatasetPostProcess(String datasetId){
+
+    // Set all depublished records back to PENDING.
+    depublishRecordIdDao.markRecordIdsWithDepublicationStatus(datasetId, null,
+            DepublicationStatus.PENDING_DEPUBLICATION, null);
+
+    // Find latest PUBLISH Type Plugin and set dataStatus to DELETED.
+    final PluginWithExecutionId<MetisPlugin> latestSuccessfulPlugin = workflowExecutionDao
+            .getLatestSuccessfulPlugin(datasetId, OrchestratorService.PUBLISH_TYPES);
+    if (Objects.nonNull(latestSuccessfulPlugin) && Objects
+            .nonNull(latestSuccessfulPlugin.getPlugin())) {
+      final WorkflowExecution workflowExecutionToUpdate = workflowExecutionDao
+              .getById(latestSuccessfulPlugin.getExecutionId());
+      final Optional<AbstractMetisPlugin> metisPluginWithType = workflowExecutionToUpdate
+              .getMetisPluginWithType(latestSuccessfulPlugin.getPlugin().getPluginType());
+      if (metisPluginWithType.isPresent()) {
+        metisPluginWithType.get().setDataStatus(DataStatus.DELETED);
+        workflowExecutionDao.updateWorkflowPlugins(workflowExecutionToUpdate);
       }
-      final Dataset dataset = datasetDao.getDatasetByDatasetId(datasetId);
-      dataset.setPublicationFitness(PublicationFitness.UNFIT);
+    }
+
+    // Set publication fitness to UNFIT.
+    final Dataset dataset = datasetDao.getDatasetByDatasetId(datasetId);
+    dataset.setPublicationFitness(PublicationFitness.UNFIT);
+    datasetDao.update(dataset);
+  }
+
+  private void depublishRecordPostProcess(DepublishPlugin depublishPlugin, String datasetId)
+          throws DpsException {
+
+    // Retrieve the successfully depublished records.
+    final long externalTaskId = Long.parseLong(depublishPlugin.getExternalTaskId());
+    final List<SubTaskInfo> subTasks = new ArrayList<>();
+    while (true) {
+      final List<SubTaskInfo> subTasksBatch = dpsClient.getDetailedTaskReportBetweenChunks(
+              depublishPlugin.getTopologyName(), externalTaskId, subTasks.size(),
+              subTasks.size() + ECLOUD_REQUEST_BATCH_SIZE);
+      subTasks.addAll(subTasksBatch);
+      if (subTasksBatch.size() < ECLOUD_REQUEST_BATCH_SIZE) {
+        break;
+      }
+    }
+
+    // Mark the records as DEPUBLISHED.
+    final Map<String, Set<String>> successfulRecords = subTasks.stream()
+            .filter(subTask -> subTask.getRecordState() == RecordState.SUCCESS)
+            .map(SubTaskInfo::getResource).map(DepublishRecordIdUtils::decomposeFullRecordId)
+            .collect(Collectors.groupingBy(Pair::getLeft,
+                    Collectors.mapping(Pair::getRight, Collectors.toSet())));
+    successfulRecords.forEach((dataset, records) ->
+            depublishRecordIdDao.markRecordIdsWithDepublicationStatus(dataset, records,
```

Comment:
I also commented on merkRecordIdsWithDepublicationStatus, why are we adding those records if missing?
I feel we should just skip in case that happens.. That shouldn't be happening in a normal scenario and if it does it will be confusing on the frontend side

Reply:
See my other comment.